### PR TITLE
Apply normalization method and exclude subjects with neuro diseases in statistical analysis

### LINKED ISSUE #76: Exclude subjects with history of neurological, cognitive or psychological diseases
## Context

Since the project aims to develop a normalization method for spinal cord CSA, the study is conducted on healthy subjects. 
The subjects presenting any history of neurological, cognitive or psychological diseases need to be excluded from the study.

## Suggestion

* Check the subjects that have any of the fields in :

    * [Nervous system disorders](https://biobank.ndph.ox.ac.uk/showcase/label.cgi?id=2406)
    * [Mental and behavioural disorders](https://biobank.ndph.ox.ac.uk/showcase/label.cgi?id=2405)

(export the fields)

Either

* Add a key in [exclude.yml](https://github.com/sct-pipeline/ukbiobank-spinalcord-csa/blob/master/exclude.yml) for neurological, cognitive or psychological diseases so we can add the concerned subjects.

or 

* Directly proceed to exclusion in [compute_stats.py](https://github.com/sct-pipeline/ukbiobank-spinalcord-csa/blob/master/pipeline_ukbiobank/cli/compute_stats.py) since we have the fields in the .csv file.

### LINKED ISSUE #77: Apply normalization of CSA  in  `compute_stats.py`
## Context
After generating a mulivariate model using a stepwise regression, the coefficients are use to generate a normalization method following:

image

## TODO
Add the following features to `compute_stats.py` :
- [x] Normalize CSA values
- [x] Compute new COV after normalization

### PR DESCRIPTION
**Pull request by sandrinebedard**
## Description
This PR addresses #77 and #76. It includes various improvements in the statistical analysis of this project.

- Apply normalization method, compute new COV.
- Exclude subjects with history of nervous system disorders.
- Have the possibility to compute stepwise linear regression with different predictors (useful to not have Brain Volume and White Volume for example both as an input --> collinearity of variables).
- Sum right and left thalamus volume .
- Add units of each parameter and fix output text of statistics.
- Add on the violin plot of CSA for sex mean and std.
- Add p-values as ** for significance of person's correlation coefficients in correlation matrix.

## Linked issues
Closes #77 and closes #76

### CODE REVIEW

**Comment by sandrinebedard** [comment]
@jcohenadad can you take a quick look at this, I want to merge this before adding new statistics and scatterplots for PMJ-based CSA 😊

File: pipeline_ukbiobank/cli/compute_stats.py [hunk 1]
```diff
@@ -31,14 +30,32 @@
 hdlr = logging.StreamHandler(sys.stdout)
 logging.root.addHandler(hdlr)
 
-PREDICTORS = ['Sex', 'Height', 'Weight', 'Age', 'Vscale', 'Volume ventricular CSF', 'Brain GM volume', 'Brain WM volume', 'Total brain volume norm', 'Total brain volume', 'Volume of thalamus (L)', 'Volume of thalamus (R)'] # TODO Add units of each
+# Dictionnary of the predictors with units
+PREDICTORS = {
+    'Sex': '',
+    'Height': 'cm',
+    'Weight': 'kg',
+    'Age': 'y.o.',
```
Comment (jcohenadad) [inline pipeline_ukbiobank/cli/compute_stats.py:38]:
```suggestion
    'Age': 'year',
```

File: pipeline_ukbiobank/cli/compute_stats.py [hunk 2]
```diff
@@ -31,14 +30,32 @@
...
+    'Vscale': '',
+    'Volume ventricular CSF': 'mm\u00b2',
```
Comment (jcohenadad) [inline pipeline_ukbiobank/cli/compute_stats.py:40]: weird characters-- maybe replace with "^"?

File: pipeline_ukbiobank/cli/compute_stats.py [hunk 3]
```diff
@@ -231,17 +256,32 @@ def compare_sex(df, path):
...
+            verticalalignment='top', horizontalalignment='left')
+    plt.ylabel('CSA (mm$^2$)')
+    fname_fig = os.path.join(path, 'violin_plot.png')
     plt.savefig(fname_fig)
 
     # Write results
     logger.info('Created: ' + fname_fig)
-    logger.info('Mean CSA value for female : {:.4} mm^2, std is {:.4}'.format(mean_csa_F, std_csa_F))
-    logger.info('Mean CSA value for male : {:.4} mm^2, std is {:.4}'.format(mean_csa_M, std_csa_M))
+    logger.info('Mean CSA value for female : {:.4} mm\u00b2, std is {:.4}'.format(mean_csa_F, std_csa_F))
```
Comment (jcohenadad) [inline pipeline_ukbiobank/cli/compute_stats.py:283]: same comment about the weird ASCII character, but i guess you have a good reason for doing so

**Review by jcohenadad — APPROVED** [review]
i cannot assess the functionality of this PR as it would require me to test it with data (and that would take me too much time, which i don't have right now)-- so i'm just going to approve and we can revise results together

File: pipeline_ukbiobank/cli/compute_stats.py (hunk 2 above)
Comment (sandrinebedard) [reply pipeline_ukbiobank/cli/compute_stats.py:40]: the goal was to display the 2 as an exponent like this:
image

But in the saved .txt file it still displays as ^2, so i'll change it back!

**Comment by sandrinebedard** [comment]
> i cannot assess the functionality of this PR as it would require me to test it with data (and that would take me too much time, which i don't have right now)-- so i'm just going to approve and we can revise results together

ok good, I'll add that to the agenda for our next meeting!

**Pull request merged**